lib/mayaUsd/render/vp2RenderDelegate/basisCurves.cpp

```diff
@@ -447,14 +447,10 @@ void HdVP2BasisCurves::Sync(
     const SdfPath& id = GetId();
     if (HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->normals)
         || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->primvar)) {
-        const HdVP2Material* material = static_cast<const HdVP2Material*>(
-            renderIndex.GetSprim(HdPrimTypeTokens->material, GetMaterialId()));
-
-        TfToken              materialNetworkToken = _GetMaterialNetworkToken(reprToken);
-        const TfTokenVector& requiredPrimvars
-            = (material && material->GetSurfaceShader(materialNetworkToken))
-            ? material->GetRequiredPrimvars(materialNetworkToken)
-            : sFallbackShaderPrimvars;
+        TfTokenVector requiredPrimvars;
+        if (!_GetMaterialPrimvars(renderIndex, GetMaterialId(), requiredPrimvars)) {
+            requiredPrimvars = sFallbackShaderPrimvars;
+        }
```

Comment:
Since primvars (and their vertex buffers) are shared among all representations, we need to include here all required primvars from all available material networks, which is done inside _GetMaterialPrimvars method.